Code/src/main/java/nl/utwente/group10/ui/components/blocks/ValueBlock.java

```diff
@@ -76,17 +77,12 @@ public final StringProperty valueProperty() {
 
     @Override
     public Expr asExpr() {
-        // TODO: support more types than floats
-        return new Value(new ConstT("Float"), getValue());
+        return new Value(type, getValue());
```

Comment:
This is correct for now, but this might require (some) changes when merged with ui-2.